spartan/releases/rough-rhino/aztec-spartan.sh

```diff
@@ -0,0 +1,285 @@
+#!/bin/bash
+
+# Colors and formatting
+BLUE='\033[0;34m'
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+# Global variables
+ARCH=$(uname -m)
+DEFAULT_P2P_PORT="40400"
+DEFAULT_PORT="8080"
+DEFAULT_KEY="0x0000000000000000000000000000000000000000000000000000000000000001"
+# Try to get default IP from ipify API, otherwise leave empty to require user input
+DEFAULT_IP=$(curl -s --connect-timeout 5 https://api.ipify.org?format=json | grep -o '"ip":"[^"]*' | cut -d'"' -f4 || echo "")
+DEFAULT_NAME="validator-1"
+
+# Parse command line arguments
+parse_args() {
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            -p|--port)
+                CLI_PORT="$2"
+                shift 2
+                ;;
+            -p2p|--p2p-port)
+                CLI_P2P_PORT="$2"
+                shift 2
+                ;;
+            -ip|--ip)
+                CLI_IP="$2"
+                shift 2
+                ;;
+            -k|--key)
+                CLI_KEY="$2"
+                shift 2
+                ;;
+            -n|--name)
+                CLI_NAME="$2"
+                shift 2
+                ;;
+            *)
+                shift
+                ;;
+        esac
+    done
+}
+
+# Show banner function
+show_banner() {
+    echo -e "${BLUE}"
+    echo "    _    ____ _____ _____ ____   _____ _____ ____ _____ _   _ _____ _____ "
+    echo "   / \  |_  /|_   _| ____|  _ \ |_   _| ____/ ___|_   _| \ | | ____|_   _|"
```

Comment:
aztec not azter?

Reply:
I'm gonna shamelessly blame it on an AI. It will come for me anyway